driver-core/src/main/com/mongodb/operation/AggregateOperation.java

```diff
@@ -352,7 +352,7 @@ public BsonDocument getHint() {
         return withConnection(binding, new CallableWithConnectionAndSource<BatchCursor<T>>() {
             @Override
             public BatchCursor<T> call(final ConnectionSource source, final Connection connection) {
-                validateReadConcernAndCollation(connection, readConcern, collation);
+                validateReadConcernAndCollation(connection, binding.getSessionContext().getReadConcern(), collation);
```

Comment:
All the operations that use read concern now get it from the session.  Once async is done, we'll remove the read concern properties in these operations as well.